server/handlers/order.js

```diff
@@ -58,3 +59,27 @@ exports.createOrderHandler = async function (req, res, next) {
 
   return res.json(Order.fromMongoJSON(result));
 };
+
+exports.getOrdersByClientIDValidator = [orderClientIDQueryValidator];
+
+exports.getOrdersByClientID = async function (req, res, next) {
+  let result;
+
+  try {
+    result = await dao.getOrdersByClientID(req.query.clientID);
+  } catch (err) {
+    console.error(
+      `getOrdersByClientID() -> couldn't retrieve client's orders: ${err}`
+    );
+    return res.status(500).end();
+  }
+
```

Comment:
You can place here something like if(!result) and return 404. then res.json with .map. I remember the template was something similar. In this way you map only if you need to return something.